velox/type/TimestampConversion.h

```diff
@@ -107,6 +107,38 @@ Status daysSinceEpochFromWeekDate(
     int32_t dayOfWeek,
     int64_t& out);
 
+/// Computes the signed number of days since the Unix epoch (1970-01-01). To
+/// align with Spark's SimpleDateFormat behavior, this function offers two
+/// modes: lenient and non-lenient. For non-lenient mode, it returns an error
+/// status if the date is invalid. For lenient mode, it accepts a wider range of
+/// arguments.
+/// @param year Year. For non-lenient mode, it should be in the range [1,
```

Comment:
> it should be in the range [1, 292278994]. e.g: 1996, -2000.

typo? The range doesn't seem to allow negative values or zero.

Reply:
Not a typo. The min value of year in non-lenient mode is 1.  Here is Java implementation:

>  static final int MIN_VALUES[] = {
        BCE,            // ERA
        1,              // YEAR
        JANUARY,        // MONTH
        1,              // WEEK_OF_YEAR
        0,              // WEEK_OF_MONTH
        1,              // DAY_OF_MONTH
        1,              // DAY_OF_YEAR
        SUNDAY,         // DAY_OF_WEEK
        1,              // DAY_OF_WEEK_IN_MONTH
        AM,             // AM_PM
        0,              // HOUR
        0,              // HOUR_OF_DAY
        0,              // MINUTE
        0,              // SECOND
        0,              // MILLISECOND
        -13*ONE_HOUR,   // ZONE_OFFSET (UNIX compatibility)
        0               // DST_OFFSET
    };

Reply:
@NEUpanning Got it. In this case "e.g: 1996, -2000." needs updating since -2000 is not allowed. 

It might be helpful to re-iterate that dates before Jan 1, 1970 are not supported in non-lenient mode.

I find it hard to say / write non-lenient. Would it make sense to flip the boolean to 'strict' mode?

Reply:
@mbasmanova I assume using `non-lenient` is clearer because it indicates we only have tow modes. If using `strict`, readers might wonder if there is a `balanced` or `moderate` mode.

Reply:
I'm not sure I follow. I was suggesting to consider replacing 'lenient' boolean with 'strict'.

Reply:
I believe other TS/DS API also use term 'strict', so it would be nice to stick to that term if possible.

Reply:
Spark uses the term 'lenient', and this function aligns with Spark. See [link](https://github.com/apache/spark/blob/master/sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala#L505). Therefore, I think 'lenient' would be more appropriate.

Reply:
@NEUpanning I assume this suggestion is about replacing 'non-lenient' with 'strict' which makes it more straightforward to understand. The 'lenient' will stay unchanged.

Reply:
@rui-mo 
> I assume using non-lenient is clearer because it indicates we only have tow modes. If using strict, readers might wonder if there is a balanced or moderate mode.

This is my concern. If this is not a problem or rarely happens, I'd like to replace it.

Reply:
@NEUpanning Since it has been clarified in the comment that there are only two modes, so I think it is fine. Thanks.

Reply:
Thanks for your suggestion. I've replace it.